apollo-runtime/src/appleMain/kotlin/com/apollographql/apollo3/internal/dispatchers.kt

```diff
@@ -1,79 +1,29 @@
 package com.apollographql.apollo3.internal
 
-import kotlinx.coroutines.CancellableContinuation
 import kotlinx.coroutines.CoroutineDispatcher
-import kotlinx.coroutines.Delay
-import kotlinx.coroutines.DisposableHandle
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.InternalCoroutinesApi
-import kotlinx.coroutines.Runnable
-import platform.Foundation.NSThread
-import platform.darwin.DISPATCH_TIME_NOW
-import platform.darwin.dispatch_after
-import platform.darwin.dispatch_async
-import platform.darwin.dispatch_get_main_queue
-import platform.darwin.dispatch_time
-import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.newFixedThreadPoolContext
+import kotlinx.coroutines.newSingleThreadContext
 
-internal actual fun defaultDispatcher(requested: CoroutineDispatcher?): CoroutineDispatcher {
-  check(requested == null || requested is DefaultDispatcher) {
-    "Changing the dispatcher is not supported on Apple targets"
-  }
-  check(NSThread.isMainThread) {
-    "defaultDispatcher must be called from the main thread"
-  }
+private val defaultDispatcher = newFixedThreadPoolContext(nThreads = 32, name = "Apollo Default Dispatcher")
```

Comment:
Could we use `Dispatchers.Default` here? This way it pools the [`global queue`](https://developer.apple.com/documentation/dispatch/1452927-dispatch_get_global_queue) and saves a bunch of memory/threads?

Reply:
Looks like it! 

(Can't remember where I (mis?)read that Dispatchers.Default and Main were the same on Apple  - nope that's fake news ❌)

Reply:
Ah found it, it was on the [doc for the Main dispatcher](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-main.html) saying _"On JS and Native it is equivalent to the Default dispatcher"_

But the doc must be outdated for the new MM.


```kotlin
        withContext(Dispatchers.Main) {
            println(NSThread.isMainThread)
        }
        withContext(Dispatchers.Default) {
            println(NSThread.isMainThread)
        }
```
prints
- `true`, `true` with the old MM
- `true`, `false` with the new MM